packages/ui-truncate-text/src/TruncateText/utils/truncate.ts

```diff
@@ -55,100 +65,80 @@ import cleanData from './cleanData'
  * @param {number} options.lineHeight=1.2 Unitless multiplier to use in case element can have
  * 'normal' lineHeight. Adjust this to better match your font if needed.
  */
-
-// @ts-expect-error ts-migrate(7030) FIXME: Not all code paths return a value.
-function truncate(element, options) {
+function truncate(element: Element, options?: TruncatorOptions) {
   const truncator = new Truncator(element, options)
   if (truncator) {
     return truncator.truncate()
   }
+  return
 }
 
 class Truncator {
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'element' implicitly has an 'any' type.
-  constructor(element, options = {}) {
-    // @ts-expect-error ts-migrate(2339) FIXME: Property '_options' does not exist on type 'Trunca... Remove this comment to see the full error message
+  private _options: Required<Omit<TruncatorOptions, 'parent'>> &
+    Pick<TruncatorOptions, 'parent'>
```

Comment:
this type is a bit tricky. I've made every field required because they are assigned a value in the constructor, except for "parent"